src/sharing/ShareWrapper.js

```diff
@@ -157,13 +154,34 @@ class ShareWrapperInner extends React.PureComponent<Props, State> {
       return;
     }
 
-    this.setSending();
+    this.setState({ sending: true });
     showToast(_('Sending message…'));
     if (sharedData.type === 'file') {
       const { files } = this.state;
       for (let i = 0; i < files.length; i++) {
-        const response = await api.uploadFile(auth, files[i].url, files[i].name);
-        messageToSend += `\n[${files[i].name}](${response.uri})\n`;
+        const { url, name: fileName } = files[i];
+        try {
+          const response = await api.uploadFile(auth, url, fileName);
+          messageToSend += `\n[${fileName}](${response.uri})\n`;
+        } catch (errorIllTyped) {
+          const error: mixed = errorIllTyped; // https://github.com/facebook/flow/issues/2470
+
+          if (!(error instanceof Error)) {
+            logging.error('ShareWrapper: Unexpected non-error thrown');
+            return;
+          }
+
+          let msg = undefined;
+          if (error instanceof ApiError && error.message.length > 0) {
+            msg = error.message;
+          }
+
+          // TODO: handle other kinds of errors
+
+          this.setState({ sending: false });
+          showErrorAlert(_('Failed to upload file: {fileName}', { fileName }), msg);
```

Comment:
How about using `error.message` for all `Error` instances, not just `ApiError`?

Reply:
Sure. The message won't always be good user-facing text, but at least it'll help us debug when we get reports.